File: pact-jvm-matchers/src/main/scala/au/com/dius/pact/matchers/BodyMatcher.scala
package au.com.dius.pact.matchers

import au.com.dius.pact.model.{BodyMismatch, HttpPart}
import au.com.dius.pact.util.Optional

trait BodyMatcher {
  def matchBody(expected: HttpPart, actual: HttpPart, allowUnexpectedKeys: Boolean) : List[BodyMismatch]

  def toScalaOption(body: Optional[String]) = {
    if (body.isPresent) Some(body.get)
    else None
  }
}

File: core/matchers/src/main/kotlin/au/com/dius/pact/core/matchers/BodyMatcher.kt
package au.com.dius.pact.core.matchers

import au.com.dius.pact.core.model.OptionalBody

interface BodyMatcher {
  fun matchBody(
    expected: OptionalBody,
    actual: OptionalBody,
    context: MatchingContext
  ): BodyMatchResult
}
